test/jdk/javax/swing/JTable/8236907/LastVisibleRow.java

```diff
@@ -0,0 +1,255 @@
+            CaptureBeforeClick();
+            testRobot.waitForIdle();
+            MouseActions();
+            testRobot.waitForIdle();
+            clearSelect();
+            testRobot.waitForIdle();
+            CaptureAfterClick();
+
+            if (!compare(bufferedImageBefore, bufferedImageAfter)) {
+                throw new RuntimeException("Test Case Failed!!");
+            }
+        } finally {
+            if (frame != null) SwingUtilities.invokeAndWait(() -> frame.dispose());
+            if (dialog != null) SwingUtilities.invokeAndWait(() -> dialog.dispose());
+        }
+    }
+
+    /***
+     *
+     * Get clickable screen point for particular row and column of a table
+     * @param row
+     * @param column
+     * @return Point
+     * @throws Exception
+     */
+    private static Point getCellClickPoint(final int row, final int column) {
+        Point result;
+
+        Rectangle rect = table.getCellRect(row, column, false);
+        Point point = new Point(rect.x + rect.width / 2,
+                rect.y + rect.height / 2);
+        SwingUtilities.convertPointToScreen(point, table);
+        result = point;
+
+        return result;
+    }
+
+    private static void createAndShowGUI() {
+        final PrintRequestAttributeSet printReqAttr = new HashPrintRequestAttributeSet();
+        printReqAttr.add(javax.print.attribute.standard.OrientationRequested.LANDSCAPE);
+        frame = new JFrame();
+        frame.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
+        Container contentPane = frame.getContentPane();
+        JPanel centerPane = new JPanel(new BorderLayout());
+        centerPane.setBorder(BorderFactory.createEmptyBorder(10, 10, 10, 10));
+        JPanel tablePaneContainer = new JPanel(new BorderLayout());
+        JPanel tablePane = new JPanel(new BorderLayout());
+        table = new JTable(new Object[][]{{"row_1_col_1", "row_1_col_2", "row_1_col_3"}, {"row_2_col_1", "row_2_col_2", "row_2_col_3"}, {"row_3_col_1", "row_3_col_2", "row_3_col_3"}, {"row_4_col_1", "row_4_col_2", "row_4_col_3"}}, new String[]{"Col1", "Col2", "Col3"});
+        //JTable table = new JTable(new Object[][] {{"row_1_col_1", "row_1_col_2", "row_1_col_3"}, {"row_2_col_1", "row_2_col_2", "row_2_col_3"}, {"row_3_col_1", "row_3_col_2", "row_3_col_3"}, {"row_4_col_1", "row_4_col_2", "row_4_col_3"}, {"row_5_col_1", "row_5_col_2", "row_5_col_3"}, {"row_6_col_1", "row_6_col_2", "row_6_col_3"}, {"row_7_col_1", "row_7_col_2", "row_7_col_3"}}, new String[] {"Col1", "Col2", "Col3"});
+        table.setPreferredSize(new Dimension(0, (table.getRowHeight() * 3)));
+
+        tablePane.add(table.getTableHeader(), BorderLayout.NORTH);
+        tablePane.add(table, BorderLayout.CENTER);
+        tablePaneContainer.add(tablePane, BorderLayout.CENTER);
+        centerPane.add(tablePaneContainer, BorderLayout.NORTH);
+        contentPane.add(centerPane, BorderLayout.CENTER);
+        frame.setSize(800, 600);
```

Comment:
It will be good to center the frame on the screen by calling setLocationRelativeTo(null) so that in CI testrun is not affected, reduce the frame size also in that case..

Reply:
Updated.